server/src/test/java/io/crate/planner/operators/EquiJoinDetectorTest.java

```diff
@@ -110,4 +110,11 @@ public void test_not_hash_join_possible_if_join_condition_refers_to_columns_from
         joinCondition = sqlExpressions.asSymbol("t1.x = t1.i");
         assertThat(EquiJoinDetector.isHashJoinPossible(JoinType.INNER, joinCondition), is(false));
     }
+
+    // tracks a bug : https://github.com/crate/crate/issues/15613
+    @Test
+    public void test_equality_expression_followed_by_case_expression() {
+        Symbol joinCondition = sqlExpressions.asSymbol("t1.a = t1.a AND CASE 1 WHEN t1.a THEN false ELSE t2.b in (t2.b) END");
```

Comment:
Shouldn't this be:
```suggestion
        Symbol joinCondition = sqlExpressions.asSymbol("t1.a = t12.b AND CASE 1 WHEN t1.a THEN false ELSE t2.b in (t2.b) END");
```
otherwise it's not an equijoin, anyway?

Reply:
@matriv This is the expression from the original issue https://github.com/crate/crate/issues/15613 , so it's ok.

Reply:
I know, I'm just wondering if the `CASE` expression is not the one that trips the `false` return.

Reply:
I evaluates to false because `t1.a = t1.a` is invalid.

Reply:
exactly so, if we do:
```
@Test
    public void test_equality_expression_followed_by_case_expression() {
        Symbol joinCondition = sqlExpressions.asSymbol("t1.a = t2.b AND CASE 1 WHEN t1.a THEN false ELSE t2.b in (t2.b) END");
        assertThat(EquiJoinDetector.isHashJoinPossible(JoinType.INNER, joinCondition), is(false));
    }
```
this fails, it detects it as true, even though we have a CASE, so to fix this you need:
```
 } else {
    context.isHashJoinPossible = false;
    return null;
}
```
So when exiting early you need to make sure to set the flag correctly, as it may have switched to `true` because some part of the expression meets the requirements.